File: src/main/java/dev/cele/asa_sm/ui/frames/ModsFrame.java
package dev.cele.asa_sm.ui.frames;

import dev.cele.asa_sm.config.SpringApplicationContext;
import dev.cele.asa_sm.dto.AsaServerConfigDto;
import dev.cele.asa_sm.dto.curseforge.ModDto;
import dev.cele.asa_sm.services.ModCacheService;

import javax.swing.*;
import java.awt.*;
import java.util.Vector;

public class ModsFrame extends JDialog {

    private boolean loading = true;
    private final ModCacheService modCacheService = SpringApplicationContext.autoWire(ModCacheService.class);
    private final Vector<ModDto> mods = new Vector<>();

    public ModsFrame(JFrame mainFrame, AsaServerConfigDto configDto) {
        super(mainFrame, "Mods", true);
        setSize(500, 500);
        setLocationRelativeTo(mainFrame);

        // create a JScrollPane
        JPanel viewPort = new JPanel();
        viewPort.setLayout(new BorderLayout());
        JScrollPane scrollPane = new JScrollPane(viewPort);
        add(scrollPane);

        // create a table model

        new Thread(() -> {
            var modDtos = modCacheService.getMods(configDto.getModIds());
            mods.removeAllElements();
            mods.addAll(modDtos);
            loading = false;
        }).start();

        setSize(400, 600);
        setResizable(true);
        setLocationRelativeTo(null);
        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
        setVisible(true);
    }
}

File: src/main/java/dev/cele/asa_sm/ui/frames/ModsFrame.kt
package dev.cele.asa_sm.ui.frames

import dev.cele.asa_sm.config.SpringApplicationContext
import dev.cele.asa_sm.dto.AsaServerConfigDto
import dev.cele.asa_sm.dto.curseforge.ModDto
import dev.cele.asa_sm.services.ModCacheService
import java.awt.BorderLayout
import java.awt.Color
import java.awt.GridBagConstraints
import java.awt.GridBagLayout
import java.util.Vector
import javax.swing.*
import javax.swing.event.TableModelListener
import javax.swing.table.TableModel
import kotlin.concurrent.thread

class ModsFrame(mainFrame: JFrame, configDto: AsaServerConfigDto): JDialog(mainFrame, "Mods", true) {

    private var loading: Boolean = true
        get() = field
        set(value) {
            field = value
        }


    val modCacheService = SpringApplicationContext.autoWire(ModCacheService::class.java)

    var mods: Vector<ModDto> = Vector()

    init {
        setSize(500, 500)
        setLocationRelativeTo(mainFrame)


        //create a JScrollPane
        val viewPort = JPanel().apply {
            layout = BorderLayout()
        }.also {
            add(JScrollPane(it))
        }

        //create a table model


        thread {
            var modDtos = modCacheService.getMods(configDto.modIds);
            mods.removeAllElements()
            mods.addAll(modDtos)
            loading = false
        }

        setSize(400, 600)
        isResizable = true
        setLocationRelativeTo(null)
        defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE;
        isVisible = true
    }

}